tests/blackbox_test.go

```diff
@@ -311,3 +338,24 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 	}
 	return nil
 }
+
+// Identify and replace $uuid<num> with correct UUID
+// Variables with matching <num> should have identical UUIDs
+func replaceUUIDVars(str string, UUIDmap map[int]string) string {
+	finalStr := str
+
+	pattern := regexp.MustCompile("\\$uuid([0-9]+)")
+	for _, match := range pattern.FindAllStringSubmatch(str, -1) {
+		num_int, _ := strconv.Atoi(match[1])
+		finalStr = strings.Replace(finalStr, match[0], getUUID(num_int, UUIDmap), 1)
+	}
+	return finalStr
+}
+
+// Format: $uuid<num> where num is key; value is uuid
+func getUUID(num int, UUIDmap map[int]string) string {
+	if _, ok := UUIDmap[num]; !ok {
+		UUIDmap[num] = strings.TrimSpace(uuid.New())
```

Comment:
same thing about the TrimSpace. I glanced at the uuid implementation and it doesn't look needed.